yaml/step.go

```diff
@@ -26,7 +29,7 @@ type (
 		Name        string                 `yaml:"name,omitempty"        json:"name,omitempty" jsonschema:"required,minLength=1,description=Unique name for the step."`
 		Parameters  map[string]interface{} `yaml:"parameters,omitempty"  json:"parameters,omitempty" jsonschema:"description=Extra configuration variables for a plugin.\nReference: https://go-vela.github.io/docs/concepts/pipeline/steps/parameters/"`
 		Privileged  bool                   `yaml:"privileged,omitempty"  json:"privileged,omitempty" jsonschema:"description=Run the container with extra privileges.\nReference: https://go-vela.github.io/docs/concepts/pipeline/steps/privileged/"`
-		Pull        bool                   `yaml:"pull,omitempty"        json:"pull,omitempty" jsonschema:"description=Automatically upgrade to the latest version of the image.\nReference: https://go-vela.github.io/docs/concepts/pipeline/steps/pull/"`
+		Pull        string                 `yaml:"pull,omitempty"        json:"pull,omitempty" jsonschema:"description=Automatically upgrade to the latest version of the image.\nReference: https://go-vela.github.io/docs/concepts/pipeline/steps/pull/"`
```

Comment:
Converting the `Pull` field to a `string` for the `github.com/go-vela/types/yaml.Step` type

Reply:
Probably worthwhile to consider updating the jsonschema description to note the available options.

Reply:
I second updating the schema for available options, not required but definitely a nice to have

Reply:
@JordanSussman @kneal I believe this feedback has been addressed 👍 

Shout out to @wass3r for the tips and insight into how to make these changes 🎉